testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/patch/StudentJsonMergePatchTest.java

```diff
@@ -0,0 +1,198 @@
+package org.jboss.resteasy.test.resource.patch;
+
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.ws.rs.HttpMethod;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit5.ArquillianExtension;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.utils.PortProviderUtil;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Tests related to JSON Merge Patch
+ * @see <a href="https://tools.ietf.org/html/rfc7386">JSON Merge Patch</a>
+ * @see <a href="https://issues.redhat.com/browse/RESTEASY-2567">RESTEASY-2567</a>
+ */
+@ExtendWith(ArquillianExtension.class)
+@RunAsClient
+public class StudentJsonMergePatchTest {
+
+    private static final Logger log = Logger.getLogger(StudentJsonMergePatchTest.class);
+
+    @ArquillianResource
+    private Deployer deployer;
+
+    static Client client;
+    static final String PATCH_DEPLOYMENT = "Patch";
+    static final Student testStudentEntity = new Student()
+                                                    .setId(2L)
+                                                    .setFirstName("Alice")
+                                                    .setSchool("school2");
+
+    @BeforeAll
+    public static void setup() {
+        client = ClientBuilder.newClient();
+    }
+
+    @AfterAll
+    public static void close() {
+        client.close();
+        client = null;
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        deployer.deploy(PATCH_DEPLOYMENT);
+        createStudentEntityOnServerAndValidate(testStudentEntity, PATCH_DEPLOYMENT);
+    }
+
+    @AfterEach
+    public void afterEach() {
+        deployer.undeploy(PATCH_DEPLOYMENT);
+    }
+
+    @Deployment(managed = false, name = PATCH_DEPLOYMENT)
+    public static Archive<?> createDeployment() {
+        WebArchive war = TestUtil.prepareArchive(PATCH_DEPLOYMENT);
+        return TestUtil.finishContainerPrepare(war, null, StudentResource.class, Student.class);
+    }
```

Comment:
Why are we not using a managed deployment?

Reply:
The deployment is using a `Map` to store entities. To make tests as independent as possible we need the `Map` to be empty for each test. So either we have to have a deployment for each test or reuse deployment with refreshed state. To achieve the initial state with empty `Map` I decided to use non-managed deployment with redeployment in `each` JUnit lifecycle.

Reply:
Hmm... ...okay. Not suggesting we need to do this, but we could add a method to clear the map. I"m not sure if it's faster to redeploy or send a request to clear the map. 

This shouldn't block this PR though IMO.

Reply:
Thanks for the feedback. This solution in my opinion is the simplest available - we don't need to make additional checks whether the map is really empty after emptying it somehow and introduce any new methods for the deployment.